src/ng/rootScope.js

```diff
@@ -101,6 +101,15 @@ function $RootScopeProvider() {
         $event.currentScope.$$destroyed = true;
     }
 
+    function cleanUpScope($scope) {
+      // Recurse bottom up
+      $scope.$$childHead && cleanUpScope($scope.$$childHead);
+      $scope.$$nextSibling && cleanUpScope($scope.$$nextSibling);
```

Comment:
this is not correct for the first call to `cleanUpScope`. we don't want to traverse into siblings of the scope being destroyed.